teehistorian/index.php

```diff
@@ -1,4 +1,50 @@
 <html>
+<?php if(isset($_SERVER['PHP_AUTH_USER'])) {
+	$auths = fopen("ddrace_auths.cfg", "r");
+	if(!$auths) {
+		exit;
+	}
+
+	$hash = null;
+	$salt = null;
+	while(!feof($auths)) {
+		$line = fgets($auths);
+		if($line[0] == '#') {
+			continue;
+		}
+
+		$tokens = explode(" ", $line);
+		if($tokens[0] !== "auth_add_p") {
+			continue;
+		}
+
+		if($tokens[1] === $_SERVER['PHP_AUTH_USER']) {
+			$hash = $tokens[3];
+			$salt = $tokens[4];
+			break;
+		}
+	}
+
+	if(!$hash)
+	{
+		header('WWW-Authenticate: Basic realm="teehistorian"');
+		header('HTTP/1.0 401 Unauthorized');
+		echo "401 Unauthorized";
+		exit;
```

Comment:
This looked weird to me so I checked and apparently this is considered best practice in PHP, oh well: https://www.php.net/manual/en/features.http-auth.php